core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.spec.ts

```diff
@@ -648,4 +652,49 @@ describe('Topic editor tab directive', function() {
     topicReinitializedEventEmitter.emit();
     expect(ctrl.initEditor).toHaveBeenCalledTimes(2);
   });
+
+  it('should call the TopicUpdateService if skillId is added in the ' +
+     'diagnostic test', fakeAsync(function() {
+    var updateSkillIdForDiagosticTestSpy = spyOn(
+      TopicUpdateService, 'updateDiagnosticTestSkills');
+    $scope.selectedSkillForDiagnosticTest = skillSummary;
+    $scope.availableSkillSummariesForDiagnosticTest = [skillSummary];
+    $scope.addSkillForDiagnosticTest();
+    $rootScope.$apply();
+    tick();
+    expect(updateSkillIdForDiagosticTestSpy).toHaveBeenCalled();
+  }));
+
+  it('should call the TopicUpdateService if any skillId is removed from the ' +
+     'diagnostic test', function() {
+    var updateSkillIdForDiagosticTestSpy = spyOn(
+      TopicUpdateService, 'updateDiagnosticTestSkills');
+    $scope.selectedSkillSummariesForDiagnosticTest = [skillSummary];
+
+    $scope.removeSkillFromDiagnosticTest(skillSummary);
+    expect(updateSkillIdForDiagosticTestSpy).toHaveBeenCalled();
+  });
+
+  it('should get eligible skill for diagnostic test selection', function() {
+    $scope.skillQuestionCountDict = {
+      skill_1: 3
+    };
+    spyOn(
+      $scope.topic,
+      'getAvailableSkillSummariesForDiagnosticTest').and.returnValue(
+      [skillSummary]
+    );
+    expect($scope.getEligibleSkillSummariesForDiagnosticTest()).toEqual(
```

Comment:
Missing pre-check. See point 2 here: https://github.com/oppia/oppia/wiki/Backend-tests#guidelines-for-writing-good-tests

Reply:
Added